MetaIndexes.cpp

```diff
@@ -564,7 +564,21 @@ void IndexShards::search (
 
 }
 
-
+void IndexShards::reconstruct (idx_t key, float * recons) const
+{
+    for (int i = 0; i < shard_indexes.size(); i++) {
+        try {
+            shard_indexes[i]->reconstruct (key, recons);
+            if (verbose)
+                printf ("key %ld found in shard %d\n", key, i);
+            return;
+        } catch (const std::out_of_range& e) {
```

Comment:
Is it accurate to assume an exception of this type being raised when the key is not found? From what I saw in one of the implementations, [`IndexPQ`](https://github.com/facebookresearch/faiss/blob/master/IndexPQ.cpp#L126), `reconstruct` might throw a `FaissException` instead.

Reply:
@Enet4 Thanks! Would it useful to catch both the exceptions and have different messages for it in verbose mode or simply replace out_of_range with FaissException for brevity?